### PR TITLE
Storage Manager v1.2

### LINKED ISSUE #93: As a user, I would like to be able to clear persistent data that were stored from previous sessions

### PR DESCRIPTION
**Pull request by keke101**
- Updated developer guide for _Storage Component_
- Minor change to `StorageManager` methods
  - Now **throws** `IOException` to be handled by caller (`Duke.java`)
- Added `shouldSave` boolean variable to `Command` class
  - This variable determines whether the program should invoke `sm.save()` to save the data to disk.
  - This variable should be invoked **for commands which change the data**. E.g. Add task
  - Each command class that inherits `Command` must specify `shouldSave` value.
- Updated JUnit for `StorageManager`
- Add clear storage command ( Closes #93 )

### CODE REVIEW

**Review by chocomango — APPROVED** [review]
Seems good, can merge and test its functionality

**Comment by gmit22** [comment]
Some checks do not seem to pass

File: src/main/java/seedu/duke/Duke.java [hunk 1]
```diff
@@ -38,17 +38,31 @@ private void run() {
...
         } catch (IOException e) {
+            Ui.showError("Unable to load the data file properly, "
+                    + "please fix the data file under data/ directory. Exiting...");
             exit(1);
+        } catch (ClassCastException e) {
+            Ui.showError("Unable to parse the data file properly, "
+                    + "please check if your data file contains the correct values. Exiting...");
         }
         Ui.showWelcomeScreen();
     }
```
Comment (gmit22) [inline src/main/java/seedu/duke/Duke.java:58]: Is the storage now taking care of all the details and storing it if the program crashes.

Comment (gmit22) [reply src/main/java/seedu/duke/Duke.java:58]: Like for example I create a project with a valid command 
eg: `project /create -title try -desc create a trial project -sd 5 -duration 30` and then due to some reason it stops working and crashes due to an error. The next time I run the program, and put the command `project /view` will it show the project created in the first place

Comment (keke101) [reply src/main/java/seedu/duke/Duke.java:58]: This PR added a `shouldSave` boolean variable to all command classes. All command classes are required to define the `shouldSave` value and I have edited all the command classes to define it. Basically, those that cause changes to the data will set `shouldSave` to true. If you look at `Duke.java`, it checks the `shouldSave` value of the command class that has executed and invokes the save if `shouldSave` is true.
In your case, the project should have been successfully created and the save happens before the crash. That is why it shows the project that was created in the first place.

Comment (gmit22) [reply src/main/java/seedu/duke/Duke.java:58]: Okay, so every time something new is added to save, it will update the json file, am i correct?

**Review by gmit22 — APPROVED** [review]
Good to merge

File: src/main/java/seedu/duke/Duke.java (hunk 1 above)
Comment (keke101) [reply src/main/java/seedu/duke/Duke.java:58]: Yeap

**Pull request merged**